tests/test-verify-volume-mount.yml

```diff
@@ -1,8 +1,8 @@
 ---
 - name: Set some facts
   set_fact:
-    storage_test_mount_device_matches: "{{ ansible_mounts|selectattr('device', 'equalto', storage_test_volume._device)|list }}"
-    storage_test_mount_point_matches: "{{ ansible_mounts|selectattr('mount', 'equalto', storage_test_volume.mount_point)|list }}"
+    storage_test_mount_device_matches: "{{ ansible_mounts|json_query('[?device==`\"{}\"`]'.format(storage_test_volume._device))}}"
+    storage_test_mount_point_matches: "{{ ansible_mounts|json_query('[?mount==`\"{}\"`]'.format(storage_test_volume.mount_point))}}"
```

Comment:
I suggest to leave the original selectattr filter in a comment above, since it is probably more readable.